app/src/androidTest/java/com/tomtom/sdk/examples/maps/MapLoadingTest.kt

```diff
@@ -0,0 +1,94 @@
+package com.tomtom.sdk.examples.maps
+
+import android.os.Build
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.By
+import androidx.test.uiautomator.UiDevice
+import androidx.test.uiautomator.UiObjectNotFoundException
+import androidx.test.uiautomator.Until
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+
+@RunWith(AndroidJUnit4::class)
+class MapLoadingTest {
+
+    @get: Rule
+    val activityRule : ActivityScenarioRule<ConfigurableMapActivity> = ActivityScenarioRule(ConfigurableMapActivity::class.java)
+
+
+    @Test
+    fun testMapShowing() {
+        val uiDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+        val mapReadySelector = By.descContains("MAP READY") //specify criteria for matching UI elements
+        val mapReadyObject = uiDevice.wait(Until.hasObject(mapReadySelector), 10000L)
+
+        if (mapReadyObject) {
+            // Object with the description "MAP READY" is found on the screen
+            assertTrue("MAP READY", true)
+        } else {
+            // Object with the description "MAP NOT READY" is not found within the given timeout
+            assertFalse("MAP NOT READY", false)
```

Comment:
Why strings and not resources or some constants?

Reply:
Which one do you recommend as better for the test class?

Reply:
I changed the strings to constants for now.